components/object_store/src/disk_cache.rs

```diff
@@ -434,6 +434,35 @@ impl Paging {
     }
 }
 
+struct CacheKey {
+    aligned_range: Range<usize>,
+    filename: String,
+}
+
+impl CacheKey {
+    pub fn new(location: &Path, aligned_range: Range<usize>) -> Self {
+        let filename = CacheKey::page_cache_name(location, &aligned_range);
+        CacheKey {
+            aligned_range,
+            filename,
+        }
+    }
+
+    pub fn get_filename(&self) -> &String {
+        &self.filename
+    }
+
+    /// Generate the filename of a page file.
+    pub fn page_cache_name(location: &Path, range: &Range<usize>) -> String {
```

Comment:
This functions should read `self.range`, `self.location`, rather than pass in new args